toolchain/check/type_structure.h

```diff
@@ -28,25 +28,19 @@ class TypeStructure : public Printable<TypeStructure> {
   // lookup query.
   auto IsCompatibleWith(const TypeStructure& other) const -> bool;
 
-  // Ordering of type structures. A higher value is a better match.
-  friend auto operator<=>(const TypeStructure& lhs, const TypeStructure& rhs)
-      -> std::weak_ordering {
-    // Higher distance is a better match, and `InfiniteDistance` is treated
-    // specially as the best possible match.
-    if (lhs.distance_to_first_symbolic_type_ !=
-        rhs.distance_to_first_symbolic_type_) {
-      if (lhs.distance_to_first_symbolic_type_ == InfiniteDistance) {
-        return std::weak_ordering::greater;
-      } else if (rhs.distance_to_first_symbolic_type_ == InfiniteDistance) {
-        return std::weak_ordering::less;
-      } else {
-        return lhs.distance_to_first_symbolic_type_ <=>
-               rhs.distance_to_first_symbolic_type_;
-      }
-    }
-    // TODO: If they have a symbolic in the same position, we could use further
-    // symbolic types to get an ordering.
-    return std::weak_ordering::equivalent;
+  // Ordering of type structures. A lower value is a better match.
+  // TODO: switch to operator<=> once we can depend on
+  // std::lexicographical_compare_three_way
```

Comment:
Why can't we depend on it?

Reply:
Because the GitHub test runners [don't support it](https://github.com/carbon-language/carbon-lang/actions/runs/14269375820/job/39998830856).

Reply:
Let's explain that a bit in the TODO so someone reading it knows what to fix/look for.

Reply:
OK, done.